pkg/build/builder/docker.go

```diff
@@ -147,12 +150,20 @@ func (d *DockerBuilder) Build() error {
 
 	if push && pushTag != "" {
 		// Get the Docker push authentication
-		pushAuthConfig, authPresent := dockercfg.NewHelper().GetDockerAuth(
-			pushTag,
+		pushAuthConfig, authPresent := dockercfg.NewHelper().GetAllDockerAuths(
 			dockercfg.PushAuthType,
 		)
+		ref, err := reference.Parse(pushTag)
+		if err != nil {
+			log.V(0).Infof("error parsing image name %s: %v", ref, err)
+		}
+		if ref.Registry == "" {
+			log.V(2).Infof("defaulting registry to docker.io for image %s", pushTag)
+			ref.Registry = "docker.io"
+		}
+		auth, _ := pushAuthConfig[ref.Registry]
```

Comment:
I think this should be:
```suggestion
		auth, authPresent := pushAuthConfig[ref.Registry]
```

Reply:
agreed